upup/pkg/fi/cloudup/openstack/cloud.go

```diff
@@ -388,8 +389,37 @@ func (c *openstackCloud) FindVPCInfo(id string) (*fi.VPCInfo, error) {
 	return nil, fmt.Errorf("openstackCloud::FindVPCInfo not implemented")
 }
 
+
+// DeleteGroup in openstack will delete servergroup, instances and ports
 func (c *openstackCloud) DeleteGroup(g *cloudinstances.CloudInstanceGroup) error {
-	return fmt.Errorf("openstackCloud::DeleteGroup not implemented")
+	grp := g.Raw.(*servergroups.ServerGroup)
+
+	err := c.DeleteServerGroup(grp.ID)
+	if err != nil {
+		return fmt.Errorf("Could not server group %q: %v", grp.ID, err)
+	}
+
+	for _, id := range grp.Members {
```

Comment:
I feel like this should be above deleting the server group, otherwise the next iteration of this call will fail.